### PR TITLE
Refactor/zoneless tests

### PR DESCRIPTION
**Pull request by jnizet**

### CODE REVIEW

File: frontend/src/app/shared/authentication.guard.spec.ts [hunk 1]
```diff
@@ -21,13 +21,13 @@ describe('AuthenticationGuard', () => {
...
     authenticationService.isAuthenticated.and.returnValue(of(true));
-    const guardResult = await firstValueFrom(TestBed.runInInjectionContext(() => authenticationGuard(null, state)));
+    const guardResult = await lastValueFrom(TestBed.runInInjectionContext(() => authenticationGuard(null, state)));
     expect(guardResult).toBeTrue();
   });
 
   it('should login if not authenticated', async () => {
     authenticationService.isAuthenticated.and.returnValue(of(false));
-    const guardResult = await firstValueFrom(TestBed.runInInjectionContext(() => authenticationGuard(null, state)));
+    const guardResult = await lastValueFrom(TestBed.runInInjectionContext(() => authenticationGuard(null, state)));
```
Comment (cexbrayat) [inline frontend/src/app/shared/authentication.guard.spec.ts:30]: OOC why this change?

File: frontend/src/app/accession-holder/edit-accession-holder/edit-accession-holder.component.spec.ts [hunk 2]
```diff
@@ -70,7 +72,7 @@ describe('EditAccessionHolderComponent', () => {
     router = TestBed.inject(Router);
     spyOn(router, 'navigate');
 
-    TestBed.createComponent(ValidationDefaultsComponent).detectChanges();
+    await TestBed.createComponent(ValidationDefaultsComponent).whenStable();
```
Comment (cexbrayat) [inline frontend/src/app/accession-holder/edit-accession-holder/edit-accession-holder.component.spec.ts:75]: 👍 I forgot to change that in diversity

File: frontend/src/app/basket/basket-content/basket-content.component.spec.ts [hunk 3]
```diff
@@ -99,8 +100,9 @@ describe('BasketContentComponent', () => {
     tester = new TestComponentTester();
   });
 
-  it('should display customer information', () => {
-    tester.detectChanges();
+  it('should display customer information', async () => {
+    tester = new TestComponentTester();
```
Comment (cexbrayat) [inline frontend/src/app/basket/basket-content/basket-content.component.spec.ts:104]: is this necessary?

File: frontend/src/app/order/edit-document/edit-document.component.spec.ts [hunk 4]
```diff
@@ -11,14 +11,15 @@ import { provideI18nTesting } from '../../i18n/mock-18n.spec';
...
 class TestComponent {
-  order = {
+  readonly order = signal({
     documents: []
-  } as DetailedOrder;
-  progress: number | null = null;
+  } as DetailedOrder);
+  readonly progress = signal<number | null>(null);
   cancelled = false;
   saved: DocumentCommand = null;
```
Comment (cexbrayat) [inline frontend/src/app/order/edit-document/edit-document.component.spec.ts:24]: nit: these should be signals too

File: frontend/src/app/rb-ngb/pagination/pagination.component.spec.ts [hunk 5]
```diff
@@ -1,21 +1,25 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
 class TestComponent {
-  page: Page<string>;
-  newPage: number = null;
+  readonly page = signal<Page<string> | undefined>(undefined);
+  readonly navigate = signal(false);
 
-  navigate = false;
+  newPage: number | null = null;
```
Comment (cexbrayat) [inline frontend/src/app/rb-ngb/pagination/pagination.component.spec.ts:22]: nit: should a signal as well

**Review by cexbrayat — APPROVED** [review]

File: frontend/src/app/shared/authentication.guard.spec.ts (hunk 1 above)
Comment (jnizet) [reply frontend/src/app/shared/authentication.guard.spec.ts:30]: I had a weird afterAll exception happening *sometimes*, and since that was the last thing I changed, I wondered if it came from the firstValueFrom. I don't thing that's it, but it doesn't change much.

File: frontend/src/app/order/edit-document/edit-document.component.spec.ts (hunk 4 above)
Comment (jnizet) [reply frontend/src/app/order/edit-document/edit-document.component.spec.ts:24]: I only use them in the tests. They're not displayed nor passed to anything.

File: frontend/src/app/rb-ngb/pagination/pagination.component.spec.ts (hunk 5 above)
Comment (jnizet) [reply frontend/src/app/rb-ngb/pagination/pagination.component.spec.ts:22]: Same here

File: frontend/src/app/order/edit-document/edit-document.component.spec.ts (hunk 4 above)
Comment (cexbrayat) [reply frontend/src/app/order/edit-document/edit-document.component.spec.ts:24]: I agree it works but I don't like to make the distinction. I'll change them
